### ISSUE TITLE
Cannot select RTL TextField's first character with tapping

### ISSUE THREAD

**Issue by alimahdiyar** [flutter/flutter#54969]
## Steps to Reproduce
Create a simple app containing an RTL TextField widget and type some RTL characters and try to select the beginning of the line **on a real touch device**. (Bug still exists on emulators when using left text-align and RTL direction and tapping right side of the TextField).

![image](https://user-images.githubusercontent.com/34129327/79476337-1a7d1280-801e-11ea-87e8-a44699579d5e.png)
I created the app with a simple main.dart file as follows:
```
import 'package:flutter/material.dart';

void main() {
  runApp(MaterialApp(
    home: new MyApp(),
  ));
}

class MyApp extends StatelessWidget {

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Center(child: Text('rtl test'))),
      body: Center(
        child: TextField(
          textDirection: TextDirection.rtl,
          minLines: 3,
          keyboardType: TextInputType.multiline,
          maxLines: null,
        ),
      ),
    );
  }
}
```
Start typing some RTL characters and tap the right edge expecting the caret to be set before the first character but it was set after the first character. Also, you can't expand your selection to the most right character when not starting from "select all".

**flutter doctor result**
[✓] Flutter (Channel master, v1.18.0-6.0.pre.42, on Linux, locale en_US.UTF-8)
    • Flutter version 1.18.0-6.0.pre.42
    • Framework revision b8bd09db21 (10 hours ago), 2020-04-15 22:25:32 -0700
    • Engine revision deef2663ac
    • Dart version 2.8.0 (build 2.8.0-dev.20.0 c9710e5059)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.0)
    • Platform android-29, build-tools 29.0.0
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

• No issues found!

**Comment by alimahdiyar** [comment]
@GaryQian Please take a look at this one

**Comment by iapicca** [comment]
Hi @alimahdiyar 
I can experience the same issue in stable as well
thank you for reporting this

**Comment by alimahdiyar** [comment]
Hi @iapicca 
I think there might be a problem with skia.

**Comment by pedromassangocode** [comment]
Easily reproducible on latest `master` channel.

<details><summary>flutter doctor</summary>

```
[✓] Flutter (Channel master, 1.26.0-2.0.pre.82, on Mac OS X 10.15.7 19H2 darwin-x64, locale en)
    • Flutter version 1.26.0-2.0.pre.82 at /Users/pedromassango/dev/SDKs/flutter_master
    • Framework revision b84d8ee2df (6 hours ago), 2020-12-18 11:59:01 +0530
    • Engine revision fd6b409f95
    • Dart version 2.12.0 (build 2.12.0-157.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/pedromassango/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 12.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.1, Build version 12A7403
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2020.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 52.0.5
    • Dart plugin version 203.5981.152

[✓] VS Code (version 1.51.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • Mac OS X 10.15.7 19H2 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 87.0.4280.88

! Doctor found issues in 1 category.
Process finished with exit code 0
```
</details>
